atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueServiceImpl.java

```diff
@@ -1438,6 +1450,18 @@ public void createTables(final Map<TableReference, byte[]> tableNamesToTableMeta
         return filteredTables;
     }
 
+    private void runSchemaMutationTask(Runnable task) {
+        try {
+            Futures.getUnchecked(schemaMutationExecutor.submit(task));
+        } catch (UncheckedExecutionException e) {
+            // If this wraps an AtlasDbDependencyException, rewrap and throw
+            if (e.getCause() instanceof AtlasDbDependencyException) {
+                throw Throwables.unwrapAndThrowAtlasDbDependencyException(e);
+            }
+            throw e;
+        }
+    }
```

Comment:
Yep, makes sense (this happens when Cassandra, an AtlasDB dependency, is down).

One last thing - JavaDoc for `createTable` / `createTables` / `dropTable` / `dropTables` in this class should be `@throws AtlasDbDependencyException` rather than `@throws IllegalStateException` after this change right?

Reply:
Yep, thanks for catching